src/main/java/com/snowplowanalytics/snowplow/tracker/emitter/SimpleEmitter.java

```diff
@@ -107,4 +113,26 @@ public void run() {
             }
         };
     }
+
+    /**
+     * Returns List of Events that are in the buffer.
+     * Always empty for SimpleEmitter
+     *
+     * @return the empty buffer
+     */
+    @Override
+    public List<TrackerEvent> getBuffer() {
+        return new ArrayList<>();
+    }
+
+    /**
+     * Customize the emitter buffer size to any valid integer greater than zero.
+     * Has no effect on SimpleEmitter
+     *
+     * @param bufferSize number of events to collect before sending
+     */
+    @Override
+    public void setBufferSize(final int bufferSize) {
+        this.bufferSize = 1;
```

Comment:
I would prefer if `int bufferSize` is not a property of AbstractEmitter, but instead make it a property of BatchEmitter only.

Then the SimpleEmitter can implement `getBufferSize()` as `return 1` and the BatchEmitter can implement it as `return this.bufferSize`.  I would even consider raising an exception if you try to setBufferSize on the SimpleEmitter.

(I appreciate this has nothing to do with the current issue)